jetty-core/jetty-compression/jetty-compression-server/src/main/java/org/eclipse/jetty/compression/server/CompressionHandler.java

```diff
@@ -259,7 +263,20 @@ public boolean handle(Request request, Response response, Callback callback) thr
                 {
                     // Collect all Accept-Encoding headers.
                     if (qualityCSV == null)
-                        qualityCSV = new QuotedQualityCSV();
+                    {
+                        HttpConfiguration httpConfiguration = request.getConnectionMetaData().getHttpConfiguration();
+                        qualityCSV = (httpConfiguration.getHttpCompliance().allows(WHITESPACE_IN_PARAMETER))
+                            ? new QuotedQualityCSV()
+                            {
+                                @Override
+                                protected void onComplianceViolation(ComplianceViolation violation)
+                                {
+                                    httpConfiguration.notifyViolation(violation, this.toString());
+                                }
+                            }
+                            : new QuotedQualityCSV();
+
```

Comment:
```suggestion
                        qualityCSV = new QuotedQualityCSV()
                        {
                            @Override
                            protected void onComplianceViolation(ComplianceViolation violation)
                            {
                                if (httpConfiguration.getHttpCompliance().allows(WHITESPACE_IN_PARAMETER))
                                    httpConfiguration.notifyViolation(violation, this.toString());
                                else
                                    super.onComplianceViolation(violation);
                            }
                        };
```

Reply:
I prefer my way as it avoids creating a capturing lambda in the most common case of not allowing the violation.